components/scream/src/share/util/scream_common_physics_functions_impl.hpp

```diff
@@ -384,6 +384,74 @@ void PhysicsFunctions<DeviceT>::calculate_mmr_from_vmr(const MemberType& team,
   });
 }
 
+template<typename DeviceT>
+KOKKOS_INLINE_FUNCTION
+void PhysicsFunctions<DeviceT>::lapse_T_for_psl(const Real& T_ground, const Real& p_ground, const Real& phi_ground,
+					        Real& lapse, Real& T_ground_tmp )
+{
+  /* 
+    Choose lapse rate and effective ground temperature to use for sea-level pressure calculation.
+    This function should only be used by calculate_psl and is separated from that function solely 
+    to improve specificity of unit testing.
+ */
+
+  using C = scream::physics::Constants<Real>;
+  constexpr Real gravit = C::gravit;
+
+  //Get preliminary surface and sea level temperature to decide on lapse rate
+  auto T_sl = T_ground + 0.0065*phi_ground/gravit; //start by assuming lapse rate is 6.5 K/km
```

Comment:
Beware that, sInce 0.0065 is a double (that's the default floating point type), you are performing calculations in double precision here. `T_sl` is a double, for instance, regardless of the precision of the other inputs.

Of course, when stuff is stored in `lapse` or `T_ground_tmp` it is converted back to `Real`, so the final precision is the normal scream one. But I don't know if you want the intermediate calculations do be _always_ in double precision.

Reply:
In general, if you want your hard coded numbers to match the default precision of scream (i.e., Real), you should use the `sp` function. E.g.,

```
auto t_sl = T_ground + sp(0.0065) * phi_ground/gravit;
```
Or even just `Real(0.0065)`, which is what the `sp` function does internally.

Note: for integer numbers, you can also avoid using the `.`. So long as they are used in arithmetic expressions involving Real's, the integer will be promoted to a Real. If, otoh, you write

```
auto blah = 2.0 * some_real;
```
Then `some_real` is promoted to double, and the type of `blah` is `double`.

Reply:
Good catch. I meant to sp that number. I'll do that.